pkg/broker/filter/filter_handler.go

```diff
@@ -353,71 +363,80 @@ func (h *Handler) getTrigger(ref path.NamespacedNameUID) (*eventingv1.Trigger, e
 	return t, nil
 }
 
-func filterEvent(ctx context.Context, triggerSpec eventingv1.TriggerSpec, event cloudevents.Event) eventfilter.FilterResult {
+func (h *Handler) filterEvent(ctx context.Context, trigger *eventingv1.Trigger, event cloudevents.Event) eventfilter.FilterResult {
 	switch {
-	case feature.FromContext(ctx).IsEnabled(feature.NewTriggerFilters) && len(triggerSpec.Filters) > 0:
-		logging.FromContext(ctx).Debugw("New trigger filters feature is enabled. Applying new filters.", zap.Any("filters", triggerSpec.Filters))
-		return applySubscriptionsAPIFilters(ctx, triggerSpec.Filters, event)
-	case triggerSpec.Filter != nil:
-		logging.FromContext(ctx).Debugw("Applying attributes filter.", zap.Any("filter", triggerSpec.Filter))
-		return applyAttributesFilter(ctx, triggerSpec.Filter, event)
+	case feature.FromContext(ctx).IsEnabled(feature.NewTriggerFilters):
+		logging.FromContext(ctx).Debugw("New trigger filters feature is enabled. Applying new filters.", zap.Any("filters", trigger.Spec.Filters))
+		filter, ok := h.filtersMap.Get(trigger)
+		if !ok {
+			logging.FromContext(ctx).Debug("Found no filters for trigger", zap.String("triggerFilterKey", fmt.Sprintf("%s.%s", trigger.Namespace, trigger.Name)))
+			return eventfilter.NoFilter
+		}
+		return filter.Filter(ctx, event)
```

Comment:
One question, if `filters` is not specified, the map won't have a filter? or will it have the "no filter" which would always pass?

Reply:
We need to think about the race condition and possible concurrency of uncorrelated events: "filter added to the map" and "request received, evaluate the filter".

As it is, it might be possible that we receive a request but "AddFunc" or "UpdateFunc" is not called

Reply:
> , if filters is not specified, the map won't have a filter? or will it have the "no filter" which would always pass?

In this case, the map won't have a filter, so `eventfilter.NoFilter` gets returned.

> We need to think about the race condition and possible concurrency of uncorrelated events

@pierDipi do you think we need to use something similar to the database concept of transactions? For example, acquire a read lock on the filters map starting when calling `filtersMap.Get()`, and similarly acquire a write lock on the filters map when the "AddFunc" or similar is called? Or are you thinking of something else?

Reply:
I was thinking something simpler like: 
- always set the filter on the map (even the "no filter")
- when there isn't a filter associated with "trigger" (aka ok is false) create a filter on the fly (rare case covering the race condition / delay in setting the filter)

Reply:
I've added what you recommended, thanks for the good idea @pierDipi !!